pkg/cache/internal/informers.go

```diff
@@ -92,6 +93,25 @@ type tracker struct {
 	Metadata     map[schema.GroupVersionKind]*Cache
 }
 
+// GetOptions provides configuration to customize the behavior when
+// getting an informer.
+type GetOptions struct {
```

Comment:
Why is this type `internal`? What can be configured ends up being part of the public API so this seems to make things needlessly complicated to me. You can use aliases to make to avoid the import loop. This applies to `type InformerGetOption func(*GetOptions)` and `func BlockUntilSynced` as well.

Reply:
I'm slightly confused by this comment. How can aliasing avoid an import loop if internal needs to see the options?

I agree perpetuating the functional options is a bit unwieldy.

I see 3 alternatives:

* split this out into a shared package to avoid the import cycle (we will probably still need type aliasing on the public package for simplicity)

* Leave the GetOptions struct internal, but do not have the internal package accept functional options (this avoids growth in the arity of the Get() function if more options are added)

* Add a boolean parameter for blocking/non-blocking (probably fine, but might not scale well if more parameters are added later)

I'll do the second option for now, LMK what you think.

Reply:
@alvaroaleman Does this approach LGTY? If so, I can add some unit tests

Reply:
To be clear, I've done the second option and have added unit tests.